src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,175 @@
+import { ext } from './extensionVariables';
+import { delay } from "./utils/delay";
+import { openUrl } from './utils/openUrl';
+import { findTableByColumnName, findTableByRowValue, getValuesByColumnName } from "./utils/tableUtil";
+
+const detectorId: string = 'LinuxContainerStartFailure';
+export enum ColumnName {
+    status = "Status",
+    message = "Message",
+    name = "Data.Name",
+    value = "Data.Value",
+    expanded = "Expanded",
+    solutions = "Solutions",
+    time = "Time",
+    instance = "Instance",
+    facility = "Facility",
+    failureCount = "FailureCount"
+}
+
+export const detectorCancelTokens: Map<string, CancellationTokenSource> = new Map();
+export async function checkLinuxWebAppDownDetector(node: SiteTreeItem, tokenSource: CancellationTokenSource): Promise<void> {
+    return await callWithTelemetryAndErrorHandling('appService.linuxWebAppDownDetector', async (context: IActionContext): Promise<void> => {
+        const deployment: DeployResult = await node.root.client.kudu.deployment.getResult('latest');
+        if (!deployment.startTime) {
+            // if there's no deployment detected, nothing can be done
+            context.telemetry.properties.cancelStep = 'noDeployResult';
+            return;
+        }
+
+        const deployResultTime: Date = new Date(deployment.startTime);
+
+        const detectorOutput: string = `Diagnosing web app "${node.root.client.siteName}" for critical errors...`;
+        ext.outputChannel.appendLine(detectorOutput);
+
+        const detectorUri: string = `${node.root.environment.resourceManagerEndpointUrl}${node.id}/detectors/${detectorId}`;
+        const requestOptions: WebResource & Partial<{ qs: queryString }> = new WebResource();
+
+        requestOptions.method = 'GET';
+        requestOptions.url = detectorUri;
+        requestOptions.qs = {
+            'api-version': "2015-08-01",
+            fId: "1",
+            btnId: "2",
+            inpId: "1",
+            val: "vscode",
+            startTime: deployResultTime.toISOString(),
+            endTime: new Date().toISOString()
+        };
+        await signRequest(requestOptions, node.root.credentials);
+
+        let detectorErrorMessage: string | undefined;
+        // wait 10 minutes for a response from the detector
+        const detectorTimeoutMs: number = Date.now() + 20 * 60 * 1000;
+        do {
+            if (tokenSource.token.isCancellationRequested) {
+                // the user cancelled the check by deploying again
+                throw new UserCancelledError();
+            }
+
+            if (Date.now() > detectorTimeoutMs) {
+                const noIssuesFound: string = `No critical issues found for web app "${node.root.client.siteName}".`;
+                ext.outputChannel.appendLine(noIssuesFound);
+                return undefined;
+            }
+            // update the new end time to be now
+            requestOptions.qs.endTime = new Date().toISOString();
```

Comment:
You have to specify end time? You can't just leave it empty to get latest?

Reply:
Yeah, I asked the team working on this and he responded:

> If you do not provide a timestamp, it takes last 1 day... I think it ignores the second's but considers timestamp up to a minute's granularity by default

It's also how the portal does it, though their end time is the current time and their start time is exactly 24 hours prior:
![image](https://user-images.githubusercontent.com/5290572/62258951-72f17e80-b3c1-11e9-94d1-4d493a870a7a.png)